src/vs/workbench/contrib/notebook/browser/contrib/editorStatusBar/editorStatusBar.ts

```diff
@@ -268,17 +305,71 @@ registerAction2(class extends Action2 {
-		const view = viewlet?.getViewPaneContainer() as IExtensionsViewPaneContainer | undefined;
+	/**
+	 * Examine the most common language in the notebook
+	 * @param notebookTextModel The notebook text model
+	 * @returns What the suggested language is for the notebook. Used for kernal installing
+	 */
+	private getSuggestedLanguage(notebookTextModel: NotebookTextModel): string | undefined {
+		const metaData = notebookTextModel.metadata;
+		let suggestedKernelLanguage: string | undefined = (metaData.custom as any)?.metadata?.language_info?.name;
+		// TODO how do we suggest multi language notebooks?
+		if (!suggestedKernelLanguage) {
+			const cellLanguages = notebookTextModel.cells.map(cell => cell.language).filter(language => language !== 'markdown');
+			// Check if cell languages is all the same
+			if (cellLanguages.length > 1) {
+				const firstLanguage = cellLanguages[0];
+				if (cellLanguages.every(language => language === firstLanguage)) {
+					suggestedKernelLanguage = firstLanguage;
+				}
+			}
+		}
+		return suggestedKernelLanguage;
+	}
 
-		const extId = KERNEL_EXTENSIONS.get(viewType);
+	/**
+	 * Given a language and notebook view type suggest a kernel for installation
+	 * @param language The language to find a suggested kernel extension for
+	 * @returns A recommednation object for the recommended extension, else undefined
+	 */
+	private getSuggestedKernelFromLanguage(viewType: string, language: string): INotebookExtensionRecommendation | undefined {
+		const recommendation = KERNEL_RECOMMENDATIONS.get(viewType)?.get(language);
+		return recommendation;
+	}
+
+	private async _showKernelExtension(
+		paneCompositePartService: IPaneCompositePartService,
+		extensionWorkbenchService: IExtensionsWorkbenchService,
+		extensionService: IExtensionService,
+		viewType: string,
+		extId?: string,
+		isInsiders?: boolean
+	) {
+		// If extension id is provided attempt to install the extension as the user has requested the suggested ones be installed
 		if (extId) {
-			view?.search(`@id:${extId}`);
-		} else {
-			const pascalCased = viewType.split(/[^a-z0-9]/ig).map(uppercaseFirstLetter).join('');
-			view?.search(`@tag:notebookKernel${pascalCased}`);
+			const extension = (await extensionWorkbenchService.getExtensions([{ id: extId }], CancellationToken.None))[0];
+			const canInstall = await extensionWorkbenchService.canInstall(extension);
+			// If we can install then install it, otherwise we will fall out into searching the viewlet
+			if (canInstall) {
+				await extensionWorkbenchService.install(
+					extension,
+					{
+						installPreReleaseVersion: isInsiders ?? false,
+						context: { skipWalkthrough: true }
+					},
+					ProgressLocation.Notification
+				);
+				await extensionService.activateByEvent(`onNotebook:${viewType}`, ActivationKind.Immediate);
+				// Await 5 sec timeout
+				await new Promise((resolve) => setTimeout(resolve, 50000));
```

Comment:
Do we need a timeout here since Jupyter will make sure controllers are ready in its activation handler?

Reply:
No we do not. That was just a remanent of testing. It is unclear if Jupyter will go this route but still should not have a timeout